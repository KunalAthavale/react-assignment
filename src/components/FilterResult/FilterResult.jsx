import React from 'react';
import PropTypes from 'prop-types';

const FilterResult = props => {
  //const showAmmunitions = () =>
  //tank.ammunition.map(ammunitionType => <span key={ammunitionType}>ammunition{ammunitionType}</span>);
  const showImages = () => props.filteredTanks.map(tank =>
    <div className="col-xs-6 col-sm-6 col-md-3 col-lg-3 text-center"><img className="img-thumbnail rounded tank-img-dimension" data-toggle="tooltip" title={tank.name} src={tank.img} />{tank.name}</div>);

  return (
    <div className="">
      <div className="text-center tags-area">
        {(props.selections.type !== null) ? <span className="selection-tags">
          {props.selections.type}
        </span> : null}
        {(props.selections.country !== null) ? <span className="selection-tags">
          {props.selections.country}
        </span> : null}
        {(props.selections.name !== null) ? <span className="selection-tags">
          {props.selections.name}
        </span> : null}
      </div>
      {(props.selections.type !== null || props.selections.country || props.selections.name !== null) ? <div className="divider"></div> : null}

      <div className="images-container row" >{showImages()}</div>
    </div>
  );
};

FilterResult.propTypes = {
  selections: PropTypes.objectOf(PropTypes.string).isRequired,
  filteredTanks: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default FilterResult;
