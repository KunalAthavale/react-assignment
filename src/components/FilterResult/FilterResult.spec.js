import React from 'react';
import { shallow } from 'enzyme';
import FilterResult from './FilterResult';

describe('FilterResult', () => {
  it('should match snapshot', () => {
    let component = shallow(<FilterResult selections={{}} filteredTanks={[]} />);
    expect(component).toMatchSnapshot();
  });
});
