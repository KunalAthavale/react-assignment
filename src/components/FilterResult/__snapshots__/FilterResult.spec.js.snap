// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterResult should match snapshot 1`] = `ShallowWrapper {}`;
